### PR TITLE
Create tableOmopSnapshot and test it

### LINKED ISSUE #40: Create tableOmopSnapshot()

### PR DESCRIPTION
**Pull request by martaalcalde**
Fixes #40

### CODE REVIEW

File: R/tableOmopSnapshot.R [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+#'library(dplyr)
+#'library(OmopSketch)
+#'
+#' cdm <- mockOmopSketch(numberIndividuals = 1000)
+#'
+#' cdm |>
+#' summariseOmopSnapshot() |>
+#' tableOmopSnapshot()
+#'}
+tableOmopSnapshot <- function(summarisedOmopSnapshot){
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:19]: I would nice to use result as first argument always in table and plot functions

File: R/tableOmopSnapshot.R [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  # Initial checks ----
+  omopgenerics::assertClass(summarisedOmopSnapshot, "summarised_result")
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:21]: validateResultArgument

File: R/tableOmopSnapshot.R [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  if(summarisedOmopSnapshot |> dplyr::tally() |> dplyr::pull("n") == 0){
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:23]: result <- result |>
  visOmopResults::filterSettings(result_type == "summarise_omop_snapshot")

File: R/tableOmopSnapshot.R [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    return(
+      summarisedOmopSnapshot |>
+        visOmopResults::splitGroup() |>
+        dplyr::select("Estimate" = "variable_name", "cdm_name") |>
+        gt::gt()
+    )
+  }
+
+  t <- dplyr::tibble(
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:34]: why is this needed?

File: R/tableOmopSnapshot.R [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+           "Description", "Documentation reference", "Holder_name", "Release date", "Source name","Source type","Version",
+           "Count", "Start date", "End date")
+  ) |>
+    dplyr::full_join(
+      summarisedOmopSnapshot |>
+        visOmopResults::splitAll(),
+      by = c("variable_name", "estimate_name")
+    )
+
+  options(warn=-1)
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:54]: maybe suppressWarnings() fits better?

File: R/tableOmopSnapshot.R [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      dplyr::mutate(estimate_value = dplyr::if_else(.data$estimate_name == "count",
+                                                  format(as.numeric(.data$estimate_value), big.mark = ",", digits = 1),
+                                                  .data$estimate_value)
+    )
+  options(warn=0)
+
+  t <- t |>
+    dplyr::select(-c("result_id", "variable_name", "variable_level", "estimate_name", "estimate_type")) |>
+    visOmopResults::formatHeader(header = "cdm_name") |>
+    visOmopResults::gtTable(groupColumn = "var", groupOrder = c("General", "CDM", "Observation period"))
```
Comment (catalamarti) [inline R/tableOmopSnapshot.R:65]: there is a formatTable function in visOmop that does the same and would help us, so we can include the type argument

**Review by catalamarti — CHANGES_REQUESTED** [review]
@martaalcalde lets discuss it as some point so you don't need to do lots of interations

File: R/tableOmopSnapshot.R (hunk 4 above)
Comment (martaalcalde) [reply R/tableOmopSnapshot.R:34]: If you visualise the plot you may understand it

Comment (martaalcalde) [reply R/tableOmopSnapshot.R:34]: Here is a plot of the outcome:
Screenshot 2024-09-16 at 21 40 02

Comment (martaalcalde) [reply R/tableOmopSnapshot.R:34]: And here a plot of the summarised result:
Screenshot 2024-09-16 at 21 41 01

**Review by catalamarti — APPROVED** [review]
